### PR TITLE
Updates for win_user module

### LINKED ISSUE #8711: win_user fails to create user
I'm running a simple playbook using the win_user module. The playbook returns success, but the user is not created.

```
- name: Add a user
  hosts: all
  gather_facts: false
  tasks:
    - name: Add User
      win_user:
        name: ansible
        password: "@ns1bl3"
```

```
TASK: [Add User] ************************************************************** 
<162.242.172.78> ESTABLISH WINRM CONNECTION FOR USER: Administrator on PORT 5986 TO 162.242.172.78
<23.253.165.14> ESTABLISH WINRM CONNECTION FOR USER: Administrator on PORT 5986 TO 23.253.165.14
<23.253.165.14> REMOTE_MODULE win_user name=ansible password=VALUE_HIDDEN
<23.253.165.14> EXEC (New-Item -Type Directory -Path $env:temp -Name "ansible-tmp-1408645070.35-223607215344291").FullName | Write-Host -Separator '';
<162.242.172.78> REMOTE_MODULE win_user name=ansible password=VALUE_HIDDEN
<162.242.172.78> EXEC (New-Item -Type Directory -Path $env:temp -Name "ansible-tmp-1408645070.35-74589131620346").FullName | Write-Host -Separator '';
<23.253.165.14> PUT /tmp/tmpuzwJbT TO C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-223607215344291\\win_user
<162.242.172.78> PUT /tmp/tmpfbbAUp TO C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-74589131620346\\win_user
<162.242.172.78> PUT /tmp/tmpqpOxpE TO C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-74589131620346\\arguments
<23.253.165.14> PUT /tmp/tmpPuuLOz TO C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-223607215344291\\arguments
<162.242.172.78> EXEC PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -File "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-74589131620346\\win_user.ps1" "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-74589131620346\\arguments"; Remove-Item "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-74589131620346\" -Force -Recurse;
<23.253.165.14> EXEC PowerShell -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -File "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-223607215344291\\win_user.ps1" "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-223607215344291\\arguments"; Remove-Item "C:\Users\Administrator\AppData\Local\Temp\ansible-tmp-1408645070.35-223607215344291\" -Force -Recurse;
changed: [demo-windows-2012r2] => {"changed": true, "user_fullname": ["ansible"], "user_name": ["ansible"], "user_path": "WinNT://WORKGROUP/DEMO-WINDOWS-20/ansible"}
changed: [demo-windows2012r2-new] => {"changed": true, "user_fullname": ["ansible"], "user_name": ["ansible"], "user_path": "WinNT://WORKGROUP/DEMO-WINDOWS201/ansible"}
```

### PR DESCRIPTION
**Pull request by cchurch**
(Resubmitting changes from https://github.com/ansible/ansible/pull/8945)

These changes allow for retrieving and updating all of the local user properties available in the _General_ and _Member Of_ tabs in Windows.  It fixes https://github.com/ansible/ansible/issues/8711 by allowing a group to be specified when creating a user.

image

image

The only backwards-incompatible changes are the following result fields that have been renamed:
- `user_name` is now `name`
- `user_fullname` is now `fullname`
- `user_path` is now `path`

### CODE REVIEW

File: windows/win_user.ps1 [hunk 1]
```diff
@@ -51,56 +55,193 @@ $result = New-Object psobject @{
...
+        If (($account_disabled -ne $null) -and ($account_disabled -ne $user_obj.AccountDisabled)) {
+            $user_obj.AccountDisabled = $account_disabled
+            $result.changed = $true
+        }
+        If (($account_locked -ne $null) -and ($account_locked -ne $user_obj.IsAccountLocked)) {
+            $user_obj.IsAccountLocked = $account_locked
+            $result.changed = $true
+        }
+        If ($groups.GetType) {
+            [string[]]$current_groups = $user_obj.Groups() | ForEach { $_.GetType().InvokeMember("Name", "GetProperty", $null, $_, $null) }
```
Comment (ibratoev) [inline windows/win_user.ps1:203]: On Windows 2008 R2 SP1 Powershell 4.0 this line fails with:

```
Exception calling "InvokeMember" with "5" argument(s): "Unknown name. (Exception from HRESULT: 0x80020006 (DISP_E_UNKNOWNNAME))
```

The parameters I am testing with are:

```
name=bob password=Password12345 groups=Administrators
```

The user is not present on the machine initially.

Comment (cchurch) [inline windows/win_user.ps1:203]: Thanks, I'll look into it.

Comment (ibratoev) [inline windows/win_user.ps1:203]: It seems you have to first save the user changes before accessing or adding it to groups.

File: windows/win_user.ps1 [hunk 2]
```diff
@@ -51,56 +55,193 @@ $result = New-Object psobject @{
...
+        If (!$user_obj.GetType) {
+            $user_obj = $adsi.Create("User", $username)
+            If ($password -ne $null) {
+                $user_obj.SetPassword($password)
+            }
+            $result.changed = $true
+        }
+        ElseIf (($password -ne $null) -and ($update_password -eq 'always')) {
+            [void][system.reflection.assembly]::LoadWithPartialName('System.DirectoryServices.AccountManagement')
+            $pc = New-Object -TypeName System.DirectoryServices.AccountManagement.PrincipalContext 'Machine', $env:COMPUTERNAME
```
Comment (ibratoev) [inline windows/win_user.ps1]: On Windows 2008 R2 SP1 Powershell 4.0 with **longer than 15 symbols computer name**:

```
The network path was not found.
```

It seems you have to use the full computer name here when the name is longer than 15 symbols. The easiest way to do that I know of is:

``` powershell
$fullPCName = [System.Net.Dns]::GetHostName()
```

**Comment by cchurch** [comment]
@ibratoev I've added changes based on your feedback, was able to reproduce then fix both issues.

**Pull request merged**